spec/requests/items_spec.rb

```diff
@@ -1,33 +1,68 @@
 require 'rails_helper'
 
-describe 'Items API', type: :request do
-  it 'returns a successful response when there are no items' do
+RSpec.describe 'Items API', type: :request do
+  describe 'GET /items', type: :request do
+    it 'returns a successful response when there are no items' do
 
-    get '/api/v1/items'
+      get '/api/v1/items'
 
-    expect(response).to have_http_status(:success)
-    expect(JSON.parse(response.body).size).to eq(0)
-  end
-  it 'returns a single item' do
-    # FactoryBot.create(:item, name: 'Raspberry', sell_in: 5, quality: 10)
-    FactoryBot.create(:item)
+      expect(response).to have_http_status(:success)
+      expect(json.size).to eq(0)
+    end
+    it 'returns a single item' do
+
+      FactoryBot.create(:item)
+
+      get '/api/v1/items'
+
+      expect(response).to have_http_status(:success)
+      expect(json.size).to eq(1)
+    end
+
+    it 'returns all the items' do
+      FactoryBot.create(:item, name: 'Raspberry', sell_in: 5, quality: 11)
+      FactoryBot.create(:item, name: 'Pineapple', sell_in: 6, quality: 12)
+      FactoryBot.create(:item, name: 'Basil', sell_in: 7, quality: 13)
+      FactoryBot.create(:item, name: 'Bananas', sell_in: 8, quality: 14)
+      FactoryBot.create(:item, name: 'Granola', sell_in: 9, quality: 15)
 
-    get '/api/v1/items'
+      get '/api/v1/items'
 
-    expect(response).to have_http_status(:success)
-    expect(JSON.parse(response.body).size).to eq(1)
+      expect(response).to have_http_status(:success)
+      expect(json.size).to eq(5)
+    end
+
+    it 'returns details for a single item' do
+      FactoryBot.create(:item)
+    end
   end
 
-  it 'returns all the items' do
-    FactoryBot.create(:item, name: 'Raspberry', sell_in: 5, quality: 11)
-    FactoryBot.create(:item, name: 'Pineapple', sell_in: 6, quality: 12)
-    FactoryBot.create(:item, name: 'Basil', sell_in: 7, quality: 13)
-    FactoryBot.create(:item, name: 'Bananas', sell_in: 8, quality: 14)
-    FactoryBot.create(:item, name: 'Granola', sell_in: 9, quality: 15)
+  describe 'GET /items/:id' do
+    let!(:items) { FactoryBot.create_list(:item, 10) }
+    let(:item_id) { items.first.id }
+    before { get "/api/v1/items/#{item_id}" }
```

Comment:
Request for change - I would not put the actual test action in a before block. Instead, do like you did for your index tests and add `get "/api/v1/items/:id` to each test

Reply:
Why ? I thought it makes the tests more dry. 
"Dont just do the thing , ask 'Why?'" - Wes Bos

Reply:
Generally, we like our tests to be "DAMP", not "DRY". https://stackoverflow.com/questions/6453235/what-does-damp-not-dry-mean-when-talking-about-unit-tests